Hippo.Core/Domain/Order.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace Hippo.Core.Domain
+{
+    public class Order
+    {
+        [Key]
+        public int Id { get; set; }
+
+        [Required]
+        public string Category { get; set; }
+        [Required]
+        [MaxLength(50)]
+        public string Name { get; set; }
+        [MaxLength(250)]
+        public string Description { get; set; }
+        [MaxLength(150)]
+        public string ExternalReference { get; set; }
+        [Required]
+        [Range(0.01, double.MaxValue)]
+        public decimal Price { get; set; }
+        [Required]
+        [Range(0.0001, double.MaxValue)]
+        public decimal Quantity { get; set; }
+        [Required]
+        [Range(1, int.MaxValue)]
+        public int Installments { get; set; }
+
+
+        public decimal Adjustment { get; set; }
+        public string AdjustmentReason { get; set; }
+        public decimal SubTotal { get; set; }
+        public decimal Total { get; set; }
+        public decimal BalanceRemaining { get; set; } // do here, or just get the total of the billings?
+        public string Notes { get; set; }
+        public string AdminNotes { get; set; }
+        public string Status { get; set; }
+
+        public decimal InstalmentAmount => Math.Round(Total / Installments, 2);
+
+        [Required]
+        public int ClusterId { get; set; }
+        public Cluster Cluster { get; set; }
+
+        [Required]
+        public int CreatedById { get; set; }
+        public User CreatedBy { get; set; }
+
+        [Required]
+        public int PrincipalInvestigatorId { get; set; }
+        public User PrincipalInvestigator { get; set; }
```

Comment:
@sprucely help me here but do we want to tie this to "Account" table or "User" table?  Also I think i'd say change the name to something not so PI-focused.  Like `OrderedFor` or `Submitter` or something.  It's just PIs for now and in farm, but let's not 100% rely on that forever

Reply:
I think User is appropriate. Account records can be deleted (not just marked inactive) by puppet sync.

Reply:
I was going to have createdBy and PrincipalInvestigator be the same user if the PI initiates the order.
Would only be different if the admin initiates it.
I could also just have one and make a note of that in the history.

Reply:
Sounds good, I'd still like to make the field name not tied to PI so explicitly.